analyzers/tests/SonarAnalyzer.UnitTest/Rules/StringFormatValidatorTest.cs

```diff
@@ -30,43 +30,50 @@ public class StringFormatValidatorTest
         [DataTestMethod]
         [DataRow(ProjectType.Product)]
         [DataRow(ProjectType.Test)]
-        public void StringFormatRuntimeExceptionFreeValidator(ProjectType projectType) =>
-            builder.AddPaths("StringFormatRuntimeExceptionFreeValidator.cs")
+        public void StringFormatValidator_RuntimeExceptionFree(ProjectType projectType) =>
+            builder.AddPaths("StringFormatValidator.RuntimeExceptionFree.cs")
                 .AddReferences(TestHelper.ProjectTypeReference(projectType))
                 .Verify();
 
         [DataTestMethod]
         [DataRow(ProjectType.Product)]
         [DataRow(ProjectType.Test)]
-        public void StringFormatTypoFreeValidator(ProjectType projectType) =>
-            builder.AddPaths("StringFormatTypoFreeValidator.cs")
+        public void StringFormatValidator_TypoFree(ProjectType projectType) =>
+            builder.AddPaths("StringFormatValidator.TypoFree.cs")
                 .AddReferences(TestHelper.ProjectTypeReference(projectType))
                 .Verify();
 
         [TestMethod]
-        public void StringFormatEdgeCasesValidator() =>
-            builder.AddPaths("StringFormatEdgeCasesValidator.cs").Verify();
+        public void StringFormatValidator_EdgeCases() =>
+            builder.AddPaths("StringFormatValidator.EdgeCases.cs").Verify();
 
 #if NET
 
         [DataTestMethod]
         [DataRow(ProjectType.Product)]
         [DataRow(ProjectType.Test)]
-        public void StringFormatRuntimeExceptionFreeValidator_CSharp11(ProjectType projectType) =>
-            builder.AddPaths("StringFormatRuntimeExceptionFreeValidator.CSharp11.cs")
+        public void StringFormatValidator_RuntimeExceptionFree_CSharp11(ProjectType projectType) =>
+            builder.AddPaths("StringFormatValidator.RuntimeExceptionFree.CSharp11.cs")
                 .AddReferences(TestHelper.ProjectTypeReference(projectType))
                 .WithOptions(ParseOptionsHelper.FromCSharp11)
                 .Verify();
 
         [DataTestMethod]
         [DataRow(ProjectType.Product)]
         [DataRow(ProjectType.Test)]
-        public void StringFormatTypoFreeValidator_CSharp11(ProjectType projectType) =>
-            builder.AddPaths("StringFormatTypoFreeValidator.CSharp11.cs")
+        public void StringFormatValidator_TypoFree_CSharp11(ProjectType projectType) =>
+            builder.AddPaths("StringFormatValidator.TypoFree.CSharp11.cs")
                 .AddReferences(TestHelper.ProjectTypeReference(projectType))
                 .WithOptions(ParseOptionsHelper.FromCSharp11)
                 .Verify();
 
+        [TestMethod]
+        public void StringFormatValidator_CSharp12() =>
```

Comment:
I think it's rather the opposite: we don't need to run on both types, since there should be no connection between C# 12 features tested in `StringFormatValidator.CSharp12.cs` and project type.
So, because running twice the same test with two different project types doesn't seem to bring any benefit, and it would rather make unit test execution longer, I have decided to only test with `ProjectType.Product`.
As for why all other tests are parameterized: I don't know why, and I don't see any clue in the implementation of the rule. I haven't changed that since it goes a bit beyond the scope of this PR, and also because I am not 100% sure about the original intent of those parameterized tests.